### PR TITLE
RUN-5171 Respect query strings on http requests

### PR DESCRIPTION
**Pull request by luiemilio**

### CODE REVIEW

File: test/launcher.test.ts [hunk 1]
```diff
@@ -70,37 +70,6 @@ describe('Launcher', () => {
                 const mockConf: OsConfig = { urlPath: 'mac/x64', manifestLocation: '', namedPipeName: '', executablePath: '' };
                 await doesntThrowAsync(async () => await download(version, location, mockConf));
             });
-
-            it('parses proxy env variable', () => {
```
Comment (rdepena) [inline test/launcher.test.ts]: Deleting tests and not adding any  ?

Comment (luiemilio) [reply test/launcher.test.ts]: I had already added one in a previous PR that covers this. Deleted the older ones that are not needed anymore.

Comment (rdepena) [reply test/launcher.test.ts]: 👍

File: src/util/http.ts [hunk 2]
```diff
@@ -21,24 +21,23 @@ export const getProxy = () => {
...
-    if (getProxyVar() && options.host !== 'localhost' && options.host.substring(0, 3) !== '127') {
+    if (getProxyVar() && parsedUrl.hostname !== 'localhost' && parsedUrl.hostname.substring(0, 3) !== '127') {
+        options = <ClientRequestArgs>{};
         const proxy = getProxy();
+
         options.host = proxy.host;
         options.port = proxy.port;
         options.path = url;
-        options.headers.Host = parsedUrl.hostname;
+        options.headers = { 'Host': parsedUrl.hostname };
```
Comment (datamadic) [inline src/util/http.ts]: did we mean to override the entire object here?

Comment (luiemilio) [reply src/util/http.ts]: That is correct. The headers object hasn't been created at this point. https://nodejs.org/api/http.html#http_http_request_options_callback

**Pull request merged**